woocommerce/woocommerce
# CYS: Refactor routing approach

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR refactors how we handle the relationship between the routing and the Sidebar. 

This work has been necessary because, for #47461, we want to keep the homepage sidebar opened even if the URL is `customize-store/assembler-hub/homepage/:pattern_category`.

This is not possible with the `Navigator` component. Also, Gutenberg, for the top-route system is moving from the `Navigator` component: https://github.com/WordPress/gutenberg/pull/60466

Slack discussion: p1717594919258669-slack-CK1QYS6P6


 Related to #47461.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Head over to WooCommerce > Home > Customize your store.
2. Click on `Add your Logo`
3. Ensure that the sidebar does a transaction and the logo settings are rendered.
4. Click on the arrow near `Add your logo`
5. Ensure that the sidebar does a transaction and the main menu is rendered.
6. Repeat 3-5 for each item menu, ensuring that the related settings are rendered.




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

CYS: Refactor routing approach.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
